flink-runtime-web/src/main/java/org/apache/flink/runtime/webmonitor/HttpRequestHandler.java

```diff
@@ -0,0 +1,131 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.webmonitor;
+
+/*****************************************************************************
+ * This code is based on the "HttpUploadServerHandler" from the
+ * Netty project's HTTP server example.
+ *
+ * See http://netty.io and
+ * https://github.com/netty/netty/blob/netty-4.0.31.Final/example/src/main/java/io/netty/example/http/upload/HttpUploadServerHandler.java
+ *****************************************************************************/
+
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.channel.SimpleChannelInboundHandler;
+import io.netty.handler.codec.http.HttpContent;
+import io.netty.handler.codec.http.HttpHeaders;
+import io.netty.handler.codec.http.HttpMethod;
+import io.netty.handler.codec.http.HttpObject;
+import io.netty.handler.codec.http.HttpRequest;
+import io.netty.handler.codec.http.LastHttpContent;
+import io.netty.handler.codec.http.QueryStringDecoder;
+import io.netty.handler.codec.http.QueryStringEncoder;
+import io.netty.handler.codec.http.multipart.DefaultHttpDataFactory;
+import io.netty.handler.codec.http.multipart.DiskFileUpload;
+import io.netty.handler.codec.http.multipart.HttpDataFactory;
+import io.netty.handler.codec.http.multipart.HttpPostRequestDecoder;
+import io.netty.handler.codec.http.multipart.HttpPostRequestDecoder.EndOfDataDecoderException;
+import io.netty.handler.codec.http.multipart.InterfaceHttpData;
+import io.netty.handler.codec.http.multipart.InterfaceHttpData.HttpDataType;
+
+import java.io.File;
+import java.util.UUID;
+
+/**
+ * Simple code which handles all HTTP requests from the user, and passes them to the Router
+ * handler directly if they do not involve file upload requests.
+ * If a file is required to be uploaded, it handles the upload, and in the http request to the
+ * next handler, passes the name of the file to the next handler.
+ */
+public class HttpRequestHandler extends SimpleChannelInboundHandler<HttpObject> {
+
+	private HttpRequest request;
+
+	private boolean readingChunks;
+
+	private static final HttpDataFactory factory = new DefaultHttpDataFactory(true); // use disk
+
+	private String requestPath;
+
+	private HttpPostRequestDecoder decoder;
+
+	private final File uploadDir;
+
+	/**
+	 * The directory where files should be uploaded.
+	 */
+	public HttpRequestHandler(File uploadDir) {
+		this.uploadDir = uploadDir;
+	}
+
+	@Override
+	public void channelUnregistered(ChannelHandlerContext ctx) throws Exception {
+		if (decoder != null) {
+			decoder.cleanFiles();
+		}
+	}
+
+	@Override
+	public void channelRead0(ChannelHandlerContext ctx, HttpObject msg) throws Exception {
+		if (msg instanceof HttpRequest) {
+			request = (HttpRequest) msg;
+			requestPath = new QueryStringDecoder(request.getUri()).path();
+			if (request.getMethod() != HttpMethod.POST) {
```

Comment:
I'm not very sure about the conventions, but only `PUT` and `POST` methods have a payload associated with them, since they're intuitively *unsafe* methods, which change some state on the server. 
Further, as for `DELETE`, the HTTP specification states that there is no defined semantics for associating bodies. https://tools.ietf.org/html/rfc7231#section-4.3.5
Since the netty server currently only processes GET, DELETE and POST, I think we can safely assume no other requests will arrive, and if they do, it'll anyway return a 404.

Your comment however makes me realize that the file upload should happen with a `PUT` request.